src/realm.h

```diff
@@ -3809,4 +3822,253 @@ RLM_API void realm_sync_session_handle_error_for_testing(const realm_sync_sessio
  */
 RLM_API void realm_register_user_code_callback_error(void* usercode_error) RLM_API_NOEXCEPT;
 
+
+typedef struct realm_mongodb_collection realm_mongodb_collection_t;
+
+typedef struct realm_mongodb_find_options {
+    realm_string_t projection_bson;
+    realm_string_t sort_bson;
+    int64_t limit;
+} realm_mongodb_find_options_t;
+
+typedef struct realm_mongodb_find_one_and_modify_options {
+    realm_string_t projection_bson;
+    realm_string_t sort_bson;
+    bool upsert;
+    bool return_new_document;
+}realm_mongodb_find_one_and_modify_options_t;
+
+typedef void (*realm_mongodb_callback_t)(realm_userdata_t userdata, realm_string_t bson,
+                                         realm_app_error_t* app_error);
+
+/**
+ *  Get mongo db collection from realm mongo db client
+ *  @param user ptr to the sync realm user of which we want to retrieve the remote collection for
+ *  @param service name of the service where the collection will be found
+ *  @param database name of the database where the collection will be found
+ *  @param collection name of the collection to fetch
+ *  @return a ptr to a valid mongodb collection if such collection exists, nullptr otherwise
+ */
+RLM_API realm_mongodb_collection_t* realm_mongo_collection_get(realm_user_t* user, const char* service,
+                                                               const char* database, const char* collection);
+
+/**
+ *  Implement find for mongodb collection
+ *  @param collection ptr to the collection to fetch from
+ *  @param filter_ejson extended json string serialization representing the filter to apply to this operation
+ *  @param options set of possible options to apply to this operation
+ *  @param data user data to pass down to this function
+ *  @param delete_data deleter for user data
+ *  @param callback to invoke with the result
+ *  @return True if completes successfully, False otherwise
+ */
+RLM_API bool realm_mongo_collection_find(realm_mongodb_collection_t* collection,
+                                         realm_string_t filter_ejson,
```

Comment:
`realm_string_t` passed with value semantics. Following the same pattern of other C APIs where the struct is passed by value and not by ptr. Rather than passing `sizeof(ptr)` the total size passed will be `sizeof(ptr) + sizeof(size_t)`